src/rdkafka_queue.c

```diff
@@ -766,7 +767,7 @@ void rd_kafka_q_fix_offsets (rd_kafka_q_t *rkq, int64_t min_offset,
 		if (unlikely(rko->rko_type != RD_KAFKA_OP_FETCH))
 			continue;
 
-		if (rko->rko_u.fetch.rkm.rkm_offset < min_offset &&
```

Comment:
I think a cleaner fix is to just move the rkm_offset assignment above the if statement, this:
`rko->rko_u.fetch.rkm.rkm_offset += base_offset;`